base_partner_merge/base_partner_merge.py

```diff
@@ -274,9 +274,9 @@ def update_records(model, src, field_model='model', field_id='res_id',
             if record.model == 'ir.property':
                 continue
 
-            field_type = proxy_model._columns.get(record.name).__class__._type
+            field_spec = proxy_model._fields.get(record.name)
 
-            if field_type == 'function':
+            if field_spec.compute:
```

Comment:
This will not work for old style `fields.function()` fields as the compute attribute will be `None`. Check #180.